apollo-router/src/plugins/telemetry/apollo.rs

```diff
@@ -104,15 +116,17 @@ pub(crate) struct SubgraphErrorConfig {
     /// Handling of errors coming from all subgraphs
     pub(crate) all: ErrorConfiguration,
     /// Handling of errors coming from specified subgraphs
-    pub(crate) subgraphs: Option<HashMap<String, ErrorConfiguration>>,
+    pub(crate) subgraphs: HashMap<String, ErrorConfiguration>,
 }
 
 #[derive(Debug, Clone, Deserialize, JsonSchema)]
-#[serde(deny_unknown_fields, default)]
+#[serde(deny_unknown_fields)]
 pub(crate) struct ErrorConfiguration {
     /// Send subgraph errors to Apollo Studio
+    #[serde(default = "default_send_errors")]
     pub(crate) send: bool,
     /// Redact subgraph errors to Apollo Studio
+    #[serde(default = "default_redact_errors")]
```

Comment:
I think this is going to change the serialization behaviour of this struct. Previously, the default for the type would have been used, [which is](https://doc.rust-lang.org/std/primitive.bool.html#impl-Default-for-bool) `false` for a `bool`.

The two functions will return `true`.

Reply:
The serde defaulting did not agree with the Default impl: https://github.com/apollographql/router/pull/3968/files/61b41cbe171c2b31c954da63ce13245409410079#diff-7737729cbb75d74b8921ae972883e90ba02b94cce0217b662822ad5183e966e7L119

The root of our issues is this: https://github.com/serde-rs/serde/issues/2622.